operator/v1/types_network.go

```diff
@@ -0,0 +1,179 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Network describes the cluster's desired network configuration
+// +k8s:openapi-gen=true
+type Network struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   NetworkSpec   `json:"spec,omitempty"`
+	Status NetworkStatus `json:"status,omitempty"`
+}
+
+// NetworkStatus is currently unused. Instead, status
+// is reported in the Network.config.openshift.io object.
+type NetworkStatus struct {
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NetworkList contains a list of Network configurations
+type NetworkList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []Network `json:"items"`
+}
+
+// NetworkSpec is the top-level network configuration object.
+type NetworkSpec struct {
+	// IP address pool to use for pod IPs.
```

Comment:
clusterNetwork is a list of ...

Reply:
Funnily enough, I modelled this after the other Kubernetes objects, who explicitly reject the convention of including the variable name in the docblock.